javascript/ql/src/semmle/javascript/frameworks/EventEmitter.qll

```diff
@@ -0,0 +1,167 @@
+import javascript
+
+module EventEmitter {
+  /** Gets the name of a method on `EventEmitter` that returns `this`. */
+  string chainableMethod() {
+    result = "off" or
+    result = "removeAllListeners" or
+    result = "removeListener" or
+    result = "setMaxListeners" or
+    result = on()
+  }
+
+  /** Gets the name of a method on `EventEmitter` that registers an event handler. */
+  string on() {
+    result = "addListener" or
+    result = "on" or
+    result = "once" or
+    result = "prependListener" or
+    result = "prependOnceListener"
+  }
+  
+  /**
+   * An instanceof of the NodeJS EventEmitter class. 
+   * Extend this class to mark something as being instanceof the EventEmitter class. 
+   */
+  abstract class EventEmitter extends DataFlow::Node {
```

Comment:
I'd suggest using the `::Range` pattern here, it's generally the safer (though more verbose) option for API modelling using abstract classes.